source/tutorials/php.rst

```diff
@@ -0,0 +1,128 @@
+
+You will see output similar to:
+
+.. code-block:: console
+
+   Distribution:      Clear Linux OS
+   Installed version: 31310
+   Version URL:       https://cdn.download.clearlinux.org/update
+   Content URL:       https://cdn.download.clearlinux.org/update
+
+
+You can create a :file:`php.ini` as follows:
+
+.. code-block:: bash
+
+   sudo mkdir -p /etc/php.d
+   sudo touch /etc/php.d/php.ini
+
+This file can be edited with any of your specific configuration requirements, and will not be overwritten when swupd performs an update. The `PHP configuration file`_ documentation has complete detail about what you can set in this file.
+
+You can verify the location of the PHP configuration files with the :command:`php --ini` command:
+
+.. code-block:: bash
+
+   php --ini
+
+You should see output like this
+
+.. code-block:: console
+
+   Configuration File (php.ini) Path: /usr/share/defaults/php/
+   Loaded Configuration File:         /usr/share/defaults/php/php.ini
+   Scan for additional .ini files in: /etc/php.d
+   Additional .ini files parsed:      (none)
+
+
+This output indicates that PHP will read the php.ini file from `/usr/share/defaults/php` and will then load any further configuration from :file:`.ini` files in `/etc/php.d/`. We will create a :file:`php.ini` file in `/etc/php.d` for our use, and allow the defaults to be read from `/usr/share/defaults/php/`.
+
+
+Install PHP extensions
+**********************
+
+PHP extensions are compiled libraries designed to enable specific functions in your PHP code. |CL| provides PHP extensions in the :file:`php-extras` bundle.  Install the bundle with swupd:
+
+.. code-block:: bash
+
+   sudo swupd bundle-add php-extras
+
+You can see the list of extensions included in the `php-extras`_ bundle on the |CL| `Store`_.
+
+
+Enable PHP extensions
+*********************
+
+To enable an installed extension we need to add it to the :file:`php.ini` for the composer to use it.
+
+#. Create the :file:`php.ini` file, with the directive to load the php-imagick extension
+
+   .. code-block:: bash
+
+      sudo echo "extension=imagick.so" >> /etc/php.d/php.ini
+
+
+No further detail is required to load the extension, but you must restart the httpd service for PHP to pick up the modification to the `/etc/php.d/php.ini` file.
+
+   .. code-block:: bash
+
+      sudo systemctl restart httpd
+
+You can verify that the imagick extension has been loaded by searching through the runtime list of loaded PHP Modules.
+
+   .. code-block:: bash
+
+      php -m | grep imagick
+
+
+.. note::
+
+   Enabling an extension only requires that it be installed, added to the php.ini file and that the httpd service is restarted. However extensions may have configuration options.  These will be documented by the extension maintainer.  The options you need can be added to the :file:`/etc/php.d/php.ini` file as described by the documentation for the extension.  Be sure to restart httpd after making changes to the file.
```

Comment:
Suggest this rewrite:
To enable an extension, you must install it, add it to the php.ini file, and restart the httpd service. However, some extensions may have configuration options, which will be documented by the extension maintainer. Add the options you need to the :file:`/etc/php.d/php.ini` file as described in the extension's documentation. Be sure to restart httpd after changing the file.

Reply:
I would remove the reference to httpd service. That assumes PHP is being used with a web server, and even more specifically apache, which doesn't have to be the case. Maybe Installed, added to php.ini, and reference in the code?

Reply:
Part of the prerequisites is to install httpd, to constrain the tutorial to a standard environment.  I can add a note after the prereqs to explain that the tutorial assumes httpd, but that the user should make adjustments according to their environment.

Reply:
great re-write, and I've also pulled the httpd reference.